src/Entities/Player.cpp

```diff
@@ -2643,6 +2643,93 @@ void cPlayer::SendBlocksAround(int a_BlockX, int a_BlockY, int a_BlockZ, int a_R
 
 bool cPlayer::PlaceBlocks(const sSetBlockVector & a_Blocks)
 {
+	// Check to see if any entity intersects any block being placed
+	class DoesIntersectBlock : public cEntityCallback
+	{
+	public:
+		const sSetBlockVector & m_Blocks;
+		cWorld * m_World;
+
+		// The distance inside the block the entity can still be.
+		const double EPSILON = 0.05;
+
+		DoesIntersectBlock(const sSetBlockVector & a_Blocks, cWorld * a_World) :
+			m_Blocks(a_Blocks)
+			, m_World(a_World)
+		{
+		}
+
+		virtual bool Item(cEntity * a_Entity) override
+		{
+			cBoundingBox EntBox(a_Entity->GetPosition(), a_Entity->GetWidth() / 2, a_Entity->GetHeight());
+			for (auto blk: m_Blocks)
+			{
+				cBlockHandler * BlockHandler = cBlockInfo::GetHandler(blk.m_BlockType);
+				int x = blk.GetX();
+				int y = blk.GetY();
+				int z = blk.GetZ();
+				cBoundingBox BlockBox = BlockHandler->GetPlacementCollisionBox(
+					m_World->GetBlock(x - 1, y, z),
+					m_World->GetBlock(x + 1, y, z),
+					m_World->GetBlock(x, y - 1, z),
+					m_World->GetBlock(x, y + 1, z),
+					m_World->GetBlock(x, y, z - 1),
+					m_World->GetBlock(x, y, z + 1)
+				);
+				BlockBox.Move(x, y, z);
+
+				// Put in a little bit of wiggle room
+				BlockBox.Expand(-EPSILON, -EPSILON, -EPSILON);
+				if (EntBox.DoesIntersect(BlockBox))
+				{
+					return true;
+				}
+			}
+			return false;
+		}
+	} Callback(a_Blocks, m_World);
+
+	// Compute the bounding box for each block to be placed
+	cBoundingBox PlacingBounds(0, 0, 0, 0, 0, 0);
+	bool HasInitializedBounds = false;
+	for (auto blk: a_Blocks)
+	{
+		cBlockHandler * BlockHandler = cBlockInfo::GetHandler(blk.m_BlockType);
+		int x = blk.GetX();
+		int y = blk.GetY();
+		int z = blk.GetZ();
+		cBoundingBox BlockBox = BlockHandler->GetPlacementCollisionBox(
+			m_World->GetBlock(x - 1, y, z),
+			m_World->GetBlock(x + 1, y, z),
+			m_World->GetBlock(x, y - 1, z),
+			m_World->GetBlock(x, y + 1, z),
+			m_World->GetBlock(x, y, z - 1),
+			m_World->GetBlock(x, y, z + 1)
+		);
```

Comment:
This looks the same as the code in the entity callback. I think it'd be better to remember the bounding boxes here and pass them to the entity callback as a `std::vector<cBoundingBox>`.